src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -846,6 +846,12 @@ export class TerminalService extends Disposable implements ITerminalService {
 		}));
 		instanceDisposables.add(instance.onDidFocus(this._onDidChangeActiveInstance.fire, this._onDidChangeActiveInstance));
 		instanceDisposables.add(instance.onRequestAddInstanceToGroup(async e => await this._addInstanceToGroup(instance, e)));
+		instanceDisposables.add(instance.onDidChangeShellType(() => this._extensionService.activateByEvent(`onTerminal:${instance.shellType}`)));
+		instanceDisposables.add(Event.runAndSubscribe(instance.capabilities.onDidAddCapability, (() => {
```

Comment:
following the same pattern as https://github.com/microsoft/vscode/blob/341f2ff2dec8ff6e4559cd0c062b99c62077055a/src/vs/workbench/api/browser/mainThreadTerminalShellIntegration.ts#L38-L44

Reply:
There's also this helper which manages the instance listeners for you:

https://github.com/microsoft/vscode/blob/02549ac3b8042e2d09e4af616573fc87a23b2844/src/vs/workbench/contrib/terminal/browser/terminalView.ts#L131-L136

Either way is fine, the `onAnyInstanceAddedCapabilityType` is a little more elegant.